runtime/criusupport/criusupport.cpp

```diff
@@ -94,7 +94,6 @@ Java_openj9_internal_criu_InternalCRIUSupport_isCRIUSupportEnabledImpl(JNIEnv *e
 		}
 #endif /* defined(LINUX) */
 	}
-	setupJNIFieldIDs(env);
```

Comment:
I thought we did this early so that by the time checkpoint() was called, we'd be sure it would work (at least from an infrastructure perspective).

Would only setting up the JNI fields if `res == JNI_TRUE` avoid the assertion?

```c
if (res == JNI_TRUE) {
    setupJNIFieldIDs(env);
}
```

Reply:
> Would only setting up the JNI fields if res == JNI_TRUE avoid the assertion?

No, we moved the `isCRIUSupportEnabledImpl` native to java.base because it is needed very early by the security providers. The classes being loaded in `setupJNIFieldIDs` are in openj9.criu module which is loaded later.

> I thought we did this early so that by the time checkpoint() was called, 

Initially we did, but more recently we have tried to de-tangle things the require libcriu.so to be laoded, and status checks like `isCRIUSupportEnabledImpl`

The `setupJNIFieldIDs` call cannot fail once openj9.criu module is loaded so we are free to move it